### PR TITLE
Introduce ECS Anywhere test

### PR DESCRIPTION
**Pull request by pdostal**
- Related ticket: [poo#105693](https://progress.opensuse.org/issues/105693)
- Verification run: [pdostal-server.suse.cz](http://pdostal-server.suse.cz/tests/14231#step/ecs_anywhere/157)
- Variables needed: `PUBLIC_CLOUD=1 PUBLIC_CLOUD_VAULT_NAMESPACE=qac PUBLIC_CLOUD_PROVIDER=EC2`

### CODE REVIEW

File: schedule/sle-micro/raw_image.yaml [hunk 1]
```diff
@@ -38,6 +38,7 @@ schedule:
   - transactional/host_config
   - '{{rt}}'
   - '{{registration}}'
+  - containers/ecs_anywhere
```
Comment (jlausuch) [inline schedule/sle-micro/raw_image.yaml:41]: this should be only scheduled for `FLAVOR=ECS-Anywhere`

File: tests/containers/ecs_anywhere.pm [hunk 2]
```diff
@@ -0,0 +1,62 @@
+# SUSE's openQA tests
+#
+# Copyright © 2021 SUSE LLC
```
Comment (jlausuch) [inline tests/containers/ecs_anywhere.pm:3]:
```suggestion
# Copyright © 2022 SUSE LLC
```

File: tests/containers/ecs_anywhere.pm [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Create VM in EC2 using aws binary
+# Maintainer: qa-c team <qa-c@suse.de>
+
+use Mojo::Base 'publiccloud::basetest';
+use registration;
+use testapi;
+use utils qw(zypper_call systemctl);
```
Comment (jlausuch) [inline tests/containers/ecs_anywhere.pm:16]:
```suggestion
use utils qw(systemctl);
```

File: tests/containers/ecs_anywhere.pm [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+use version_utils 'is_sle';
+use transactional qw(process_reboot trup_call);
+use registration qw(add_suseconnect_product get_addon_fullname);
+
+sub run {
+    my ($self, $args) = @_;
+    $self->select_serial_terminal;
+
+    systemctl 'enable --now docker.service';
```
Comment (jlausuch) [inline tests/containers/ecs_anywhere.pm:25]: maybe we could call here `install_docker_when_needed` once https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/14594 is merged

File: tests/containers/ecs_anywhere.pm [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+    # Initialize the AWS provider
+    set_var 'PUBLIC_CLOUD_PROVIDER' => 'EC2';
```
Comment (jlausuch) [inline tests/containers/ecs_anywhere.pm:28]: we could also set it in group yaml

File: tests/containers/ecs_anywhere.pm [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+    assert_script_run 'echo "ECS_CLUSTER=ecsExternalInstanceRole" >> /etc/ecs/ecs.config';
+    assert_script_run 'mkdir /var/lib/ecs';
+    assert_script_run 'echo "AWS_DEFAULT_REGION=$AWS_DEFAULT_REGION" >> /var/lib/ecs/ecs.config';
+    assert_script_run 'echo "ECS_EXTERNAL=true" >> /var/lib/ecs/ecs.config';
+    systemctl 'enable --now amazon-ecs.service';
+    systemctl 'status amazon-ecs.service';
+    assert_script_run 'aws ecs list-container-instances --cluster jose-test';
+
+    # Debug
+    assert_script_run 'journalctl --no-pager -u amazon-ecs.service';
```
Comment (jlausuch) [inline tests/containers/ecs_anywhere.pm:52]: can we upload this as a log using `upload_logs`?

**Comment by pdostal** [comment]
> This is great Pavel! Some small suggestions.

Thank you

> I guess it should be fine to merge it after all and label the test with the bsc you created.

I would wait until we have at least fix attempt.

**Pull request merged**